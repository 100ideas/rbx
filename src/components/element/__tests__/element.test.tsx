import React from "react";
import renderer from "react-test-renderer";
import { Element } from "..";

describe("Element component", () => {
  it("Should Exist", () => {
    expect(Element).toMatchSnapshot();
  });

  it("Should have helpers classnames", () => {
    const component = renderer.create(
      <Element textColor={"white" as "white"} pull={"left" as "left"}>
        Facebook
      </Element>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("Should accept a react Element as renderAs prop", () => {
    const Custom = (
      props: React.ComponentProps<"p"> & { children: React.ReactNode },
    ) => (
      <p {...props}>
        Custom
        {props.children}
      </p>
    );

    const component = renderer.create(
      <Element renderAs={Custom}>This should be a p element</Element>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
