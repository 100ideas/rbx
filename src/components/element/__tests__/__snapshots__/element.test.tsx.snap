// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Element component Should Exist 1`] = `
Object {
  "$$typeof": Symbol(react.forward_ref),
  "defaultProps": Object {
    "renderAs": "div",
  },
  "render": [Function],
}
`;

exports[`Element component Should accept a react Element as renderAs prop 1`] = `
<p>
  Custom
  This should be a p element
</p>
`;

exports[`Element component Should have helpers classnames 1`] = `
<div
  className="has-text-white is-pulled-left"
>
  Facebook
</div>
`;
